x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -98,51 +54,36 @@ export function ServiceOverviewTransactionsTable(props: Props) {
     sort: DEFAULT_SORT,
   });
 
+  const { pageIndex, sort } = tableOptions;
+
+  const { transactionType } = useApmServiceContext();
   const {
-    data = {
-      totalItemCount: 0,
-      items: [],
-      tableOptions: {
-        pageIndex: 0,
-        sort: DEFAULT_SORT,
-      },
-    },
-    status,
-  } = useFetcher(
+    uiFilters,
+    urlParams: { start, end, latencyAggregationType },
+  } = useUrlParams();
+
+  const { data = INITIAL_STATE, status } = useFetcher(
     (callApmApi) => {
       if (!start || !end || !latencyAggregationType || !transactionType) {
         return;
       }
-
       return callApmApi({
         endpoint:
-          'GET /api/apm/services/{serviceName}/transactions/groups/overview',
+          'GET /api/apm/services/{serviceName}/transactions/groups/primary_statistics',
         params: {
           path: { serviceName },
           query: {
             start,
             end,
             uiFilters: JSON.stringify(uiFilters),
-            size: PAGE_SIZE,
-            numBuckets: 20,
-            pageIndex: tableOptions.pageIndex,
-            sortField: tableOptions.sort.field,
-            sortDirection: tableOptions.sort.direction,
             transactionType,
             latencyAggregationType: latencyAggregationType as LatencyAggregationType,
           },
         },
       }).then((response) => {
         return {
-          items: response.transactionGroups,
-          totalItemCount: response.totalTransactionGroups,
-          tableOptions: {
-            pageIndex: tableOptions.pageIndex,
-            sort: {
-              field: tableOptions.sort.field,
-              direction: tableOptions.sort.direction,
-            },
-          },
+          requestId: uuid(),
```

Comment:
Can we add the request id in `useFetcher`? That feels like a better place to do this.

Reply:
requestId was removed I'm using `preservePreviousData` as you suggested.